release/mk-release-series.sh

```diff
@@ -38,8 +38,9 @@ fi
 ## Update "master" branch code
 #
   cd $SQUID_VCS_PATH/squid-$NEWVER
+  git checkout -b v${NEWVER}-release-prep &&
+  git push -u origin v${NEWVER}-release-prep
```

Comment:
I strongly recommend adding error detection/handling, at least for any major new/modified script commands like these ones, especially if the script is going to automatically create official pull requests based on those commands! "Detect/handle errors" is a basic development principle. It should be the default!

Reply:
Which type of errors? from which process? with what handling?

FYI;
 * this script is run by a human who is watching the stderr / stdout display and looking for errors.
 * `origin` is the squidadm/squid repository. Intentionally different from the squid-cache/squid repository so that automation errors can be reverted with ease.
 * likewise the `$SQUID_VCS_PATH/squid-$NEWVER` and `$SQUID_WWW_PATH/` are staging locations which are published at a later time only if not manually reverted.

Reply:
> Which type of errors?

Any and all outcomes that `git` commands in question define as an error by exiting with a non-zero exit code.

> from which process? 

In this specific case, from the `git... && git...` shell command this change request is attached to. Personally, I would split that command into two `git` commands and handle errors from each one of them, but that improvement is not _required_ to address this particular change request:

```shell
git checkout -b v${NEWVER}-release-prep || exit $?
git push -u origin v${NEWVER}-release-prep || exit $?
```

> with what handling?

I recommend exiting with a non-zero error code (preferably the one returned by the failed command).

Detecting and handling errors is a well-established basic development practice. Two core developers should not be arguing whether to do it!

Reply:
Thank you. But no, an `exit` is the worst thing this particular block of code could do. All of the statements in this `if ... fi` block can be skipped by the release process and performed manually later without causing major problems. As demonstrated by the status quo code leaving the PR for an entirely separate asynchronous process to do.
 Whereas, an `exit` will leave the staging environment in a broken partially constructed state - worse than attempting and skipping this non-urgent task if it fails.

> Detecting and handling errors is a well-established basic development practice. Two core developers should not be arguing whether to do it!

Indeed. FWIW, the face-value implications of that statement are quite insulting to one or other of us. I will assume that a) you did not actually mean it, and b) we both have enough expertise to see the error handling present in this syntax.

For clarity, IMO the disagreement we have is about the specific mechanism(s) implemented to detect and handle errors in this script. We differ in both opinion + expertise and that is fine. Some of your other suggestions/requests have been useful and I have or will implement. This suggestion on this particular piece of code is a "wont do" for the reasons outlined above.

Reply:
> > Which type of errors?
> 
> Any and all outcomes that `git` commands in question define as an error by exiting with a non-zero exit code.
> 
> > from which process?
> 
> In this specific case, from the `git... && git...` shell command this change request is attached to. Personally, I would split that command into two `git` commands and handle errors from each one of them, but that improvement is not _required_ to address this particular change request:
> 
> ```shell
> git checkout -b v${NEWVER}-release-prep || exit $?
> git push -u origin v${NEWVER}-release-prep || exit $?
> ```
> 
> > with what handling?
> 
> I recommend exiting with a non-zero error code (preferably the one returned by the failed command).

This can be more easily achieved globally by using
```
set -eu -o pipefail
```
at the beginning of the script

> Detecting and handling errors is a well-established basic development practice. Two core developers should not be arguing whether to do it!

In my understanding the discussion is not about whether errors should be detected - they are. The discussion is how to achieve that.

Reply:
> In my understanding the discussion is not about whether errors should be detected - they are.

This change request is attached to one place where an error is not detected (in newly added code) but should be. Frankly, I do not understand how one can look at this change request (in its natural context) and develop an understanding that errors are detected.

> This can be more easily achieved globally by using `set -eu -o pipefail`

I agree that `set -e` is much "easier", but some bash enthusiast recommend [avoiding](http://mywiki.wooledge.org/BashFAQ/105) `set -e`. This is not my area of expertise, but their arguments sound rather convincing to me. I do not recommend using that option as the solution.

Reply:
Latest iteration of this script includes a lot more error catching and a "nuclearFallout" handler to purge all staged changes that can automatically be purged. Most of this was just tested with v7 branching, next use/test of the script will be in 2025 so I hope we can accept this and move on - with followup at that time when re-tested for v8.

Reply:
> I hope we can accept this and move on

If "this" is "this PR", then sure! I was not blocking this PR before, and I am still not blocking it now, despite my concerns about the script approach, the script quality, and the manner this PR has been handled. I have given up fighting this windmill.